File: other/560/560.go
package main

import (
	"fmt"
)

func subarraySum(nums []int, k int) int {
    pre, count := 0, 0
    mp := map[int]int{0 : 1}
    for i := 0; i < len(nums); i++ {
        pre += nums[i]
        diff := pre - k
        if _, ok := mp[diff]; ok {
            count += mp[diff]
        }
        mp[pre] += 1
    }
    return count
}

func main() {
	nums := []int{1, 1, 1}
	fmt.Println(subarraySum(nums, 2))
	nums = []int{1, 2, 3, 3}
	fmt.Println(subarraySum(nums, 3))
}

File: other/560/560.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int count = 0, pre = 0;
        unordered_map<int, int> mp{{0,1}};
        for (int i = 0; i < nums.size(); ++i) {
            pre += nums[i];
            int diff = pre - k;
            if (mp.count(diff)) {
                count += mp[diff];
            }
            ++mp[pre];
        }
        return count;
    }
};

int main() {
	vector<int> vec = {1, 1, 1};
	Solution sol;
	cout << sol.subarraySum(vec, 2) << endl;
	vec = {1, 2, 3};
	cout << sol.subarraySum(vec, 3) << endl;
}